step-templates/file-system-create-file.json

```diff
@@ -1,53 +1,59 @@
   "ActionType": "Octopus.Script",
-  "Version": 2,
+  "Version": 3,
+  "CommunityActionTemplateId": null,
   "Properties": {
-    "Octopus.Action.Script.ScriptBody": "$filePath = $OctopusParameters['FilePath']\n$fileContent = $OctopusParameters['FileContent']\n$encoding = $OctopusParameters['Encoding']\n\nNew-Item -ItemType file -Path $filePath -Value '' -force\n\nif(![string]::IsNullOrEmpty($fileContent))\n{\n  Set-Content -path $filePath -value $fileContent -encoding $encoding\n}",
-    "Octopus.Action.Script.Syntax": "PowerShell"
+    "Octopus.Action.Script.ScriptBody": "$fileName = $OctopusParameters['FileName']\n$path = $OctopusParameters['Path']\n$content = $OctopusParameters['Content']\n\nif ($fileName -and $path) {\n\n if(-Not(Test-Path $path))\n    {\n       Write-Error \"$path does not exist\"\n       exit 1\n    }\n\n    $fullPath=Join-Path $path -ChildPath $fileName\n    \n    Write-Host \"File Path: $fullPath\"\n    \n    New-Item -type file -Path $fullPath -force\n\n    if($content){\n       Add-Content $fullPath $content\n    }\n\n    Write-Host \"$fileName created at $path\"\n}\nelse{\n    Write-Error \"File creation failed : Incorrect File name/path.\"\n    exit 1\n}\n",
+    "Octopus.Action.Script.Syntax": "PowerShell",
+    "Octopus.Action.Script.ScriptSource": "Inline",
+    "Octopus.Action.RunOnServer": "false",
+    "Octopus.Action.Script.ScriptFileName": null,
+    "Octopus.Action.Package.FeedId": null,
+    "Octopus.Action.Package.PackageId": null
   },
-  "SensitiveProperties": {},
   "Parameters": [
     {
-      "Name": "FilePath",
-      "Label": "The full path where the file will be created.",
-      "HelpText": "Provide the entire physical path to which the physical file will be created. For example, 'C:\\Temp\\MyFile.txt' will create am empty text file named 'MyFile.txt' in the 'Temp' folder on the C: drive.",
-      "DefaultValue": null,
+      "Id": "607c3a21-a0ab-4b0d-aa01-0864bf2ae909",
+      "Name": "FileName",
+      "Label": "File name with extension to create on the file system.",
+      "HelpText": "Valid file name to be created under given path.",
+      "DefaultValue": "",
       "DisplaySettings": {
         "Octopus.ControlType": "SingleLineText"
-      }
+      },
+      "Links": {}
     },
     {
-      "Name": "FileContent",
-      "Label": "File Content",
-      "HelpText": "The text you would like inside the file, if nothing then will create an empty file.",
-      "DefaultValue": null,
+      "Id": "0e342c97-2613-4e13-9c1e-193666eff1c0",
+      "Name": "Path",
+      "Label": "Folder path under which specified file to be created.",
+      "HelpText": "Specify folder path, where you wish to create specified file.",
+      "DefaultValue": "",
       "DisplaySettings": {
-        "Octopus.ControlType": "MultiLineText"
-      }
+        "Octopus.ControlType": "SingleLineText"
+      },
+      "Links": {}
     },
     {
-      "Id": "43aff0e0-e7ad-40cb-ae54-a49ca03adfc5",
-      "Name": "Encoding",
```

Comment:
Any particular reason you've removed the `Encoding` parameter? Is it no longer needed?